### PR TITLE
feat: Added ability to rollback a helm release

### PR DESCRIPTION
**Pull request by manish-agrawal-ai**
# Description
Functionality added to rollback a helm release

Fixes # https://github.com/devtron-labs/devtron/issues/1281

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
New API hit

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: pkg/helmClient/client.go [hunk 1]
```diff
@@ -453,4 +459,28 @@ func (c *HelmClient) install(ctx context.Context, spec *ChartSpec) (*release.Rel
...
+
+	mergeRollbackOptions(spec, client)
+
+	client.Version = version
+
+	return client.Run(spec.ReleaseName)
+}
+
+// mergeRollbackOptions merges values of the provided chart to helm rollback options used by the client.
+func mergeRollbackOptions(chartSpec *ChartSpec, rollbackOptions *action.Rollback) {
```
Comment (nishant-d) [inline pkg/helmClient/client.go:477]: attach function with class

**Review by nishant-d — CHANGES_REQUESTED** [review]

File: pkg/service/HelmAppServiceWrapper.go [hunk 2]
```diff
@@ -214,6 +214,23 @@ func (impl *ApplicationServiceServerImpl) IsReleaseInstalled(ctx context.Context
 	return res, err
 }
 
+func (impl *ApplicationServiceServerImpl) RollbackRelease(ctx context.Context, in *client.RollbackReleaseRequest) (*client.BooleanResponse, error) {
+	releaseIdentifier := in.ReleaseIdentifier
+	impl.Logger.Infow("Rollback release request", "clusterName", releaseIdentifier.ClusterConfig.ClusterName, "releaseName", releaseIdentifier.ReleaseName,
```
Comment (vikramdevtron) [inline pkg/service/HelmAppServiceWrapper.go:219]: can we change, info > debug

File: pkg/helmClient/client.go (hunk 1 above)
Comment (manish-agrawal-ai) [reply pkg/helmClient/client.go:477]: This is utility function.

File: pkg/service/HelmAppServiceWrapper.go [hunk 3]
```diff
@@ -214,6 +214,23 @@ func (impl *ApplicationServiceServerImpl) IsReleaseInstalled(ctx context.Context
...
+		"namespace", releaseIdentifier.ReleaseNamespace, "version", in.Version)
+
+	success, err := impl.HelmAppService.RollbackRelease(in)
+	if err != nil {
+		impl.Logger.Errorw("Error in Rollback release request", "err", err)
+	}
+	impl.Logger.Info("Rollback release request served")
```
[the remark this answers is not in the document]
Comment (manish-agrawal-ai) [reply pkg/service/HelmAppServiceWrapper.go:226]: Will discuss this

File: pkg/service/HelmAppServiceWrapper.go (hunk 2 above)
Comment (manish-agrawal-ai) [reply pkg/service/HelmAppServiceWrapper.go:219]: Will discuss this

File: pkg/helmClient/client.go (hunk 1 above)
Comment (pghildiyal) [reply pkg/helmClient/client.go:477]: this is not merge, this is copy

Comment (manish-agrawal-ai) [reply pkg/helmClient/client.go]: done.

**Review by nishant-d — APPROVED** [review]

**Pull request merged**